integration/docker_integration_test_executor.go

```diff
@@ -113,15 +115,35 @@ func givenThisContainer(t *testing.T, bundle IntegrationTestDockerExecRunnerBund
 		}
 	}
 
-	err = testRunner.Runner.RunExecutable("docker", "cp", "piper-command-wrapper.sh", testRunner.ContainerName+":/piper-wrapper")
+	setupPiperBinary(t, testRunner, localPiper)
+
+	return testRunner
+}
+
+// Generate a random container name so we can start a new one for each test method
+// We don't rely on docker's random name generator for two reasons
+// First, it is easier to save the name here compared to getting it from stdout
+// Second, the common prefix allows batch stopping/deleting of containers if so desired
+// The test code will not automatically delete containers as they might be useful for debugging
+func generateContainerName() string {
+	var seededRand = rand.New(rand.NewSource(time.Now().UnixNano()))
+	return "piper-integration-test-" + strconv.Itoa(seededRand.Int())
+}
+
+func setupPiperBinary(t *testing.T, testRunner IntegrationTestDockerExecRunner, localPiper string) {
```

Comment:
Perhaps document here that the wrapper script is copied in order to capture the logs.